platforms/ide/ide-native/src/main/groovy/org/gradle/ide/visualstudio/tasks/GenerateProjectFileTask.java

```diff
@@ -74,7 +74,7 @@ public Object call() throws Exception {
     }
 
     @Internal
-    public Transformer<String, File> getTransformer() {
+    public Transformer<@org.jetbrains.annotations.NotNull String, File> getTransformer() {
```

Comment:
This is annotating with `@NotNull`, which is more restricted and does not change the result of that line.

Reply:
Ahh. Didn't notice that. I assumed the annotations were all `@Nullable`

Reply:
Why do we need to be explicit? Does it not default to `@NotNull`?

Reply:
It's possible we don't need to be explicit. It depends on how exactly Kotlin and other tools handle the intersection of JSR 305 nullability annotations (`@NonNullApi`), JetBrains nullability annotations (on `Transformer` and in the type parameters), and the potential lack of any defined default for the type parameter (`@NonNullApi` is only defined as default for `METHOD` and `PARAMETER`.). I would rather simply be explicit everywhere rather than need to track the exact state of this. In the future, under jspecify, we would omit this as `@NullMarked` would [define strict rules in this case](https://jspecify.dev/docs/user-guide/#defining-generics).